lib/node_modules/@stdlib/array/base/count-ifs/docs/types/index.d.ts

```diff
@@ -0,0 +1,149 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/// <reference types="@stdlib/types"/>
+
+import { Collection, AccessorArrayLike } from '@stdlib/types/array';
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @returns boolean indicating whether an element passes a test
+*/
+type Nullary = () => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @returns boolean indicating whether an element passes a test
+*/
+type Unary<U> = ( value: U ) => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @param index - current array element index
+* @returns boolean indicating whether an element passes a test
+*/
+type Binary<U> = ( value: U, index: number ) => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @param index - current array element index
+* @param arr - input array
+* @returns boolean indicating whether an element passes a test
+*/
+type Ternary<U> = ( value: U, index: number, arr: Collection<U> | AccessorArrayLike<U> ) => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @param index - current array element index
+* @param arr - input array
+* @returns boolean indicating whether an element passes a test
+*/
+type Predicate<U> = Nullary | Unary<U> | Binary<U> | Ternary<U>;
+
+/**
+* Counts the number of elements in the input arrays which pass the tests implemented by the predicate functions.
+*
+* @param x - input array
+* @param predicate - predicate function
+* @returns result
+*
+* @example
+* function predicate( v ) {
+*     return v > 0;
+* }
+*
+* var x = [ 0, 1, 0, 1, 1 ];
+*
+* var n = countIfs( x, predicate );
+* // returns 3
+*/
+declare function countIfs<U = unknown>( x: Collection<U> | AccessorArrayLike<U>, predicate: Predicate<U> ): number;
+
+/**
+* Counts the number of elements in the input arrays which pass the tests implemented by the predicate functions.
+*
+* @param x0 - first input array
+* @param predicate0 - first predicate function
+* @param x1 - second input array
+* @param predicate1 - second predicate function
+* @returns result
+*
+* @example
+* function predicate0( v ) {
+*     return v > 0;
+* }
+*
+* function predicate1( v ) {
+*     return v < 0;
+* }
+*
+* var x0 = [ 0, 1, 0, 1, 1 ];
+* var x1 = [ -1, 2, 4, -5, -8 ];
+*
+* var n = countIfs( x0, predicate0, x1, predicate1 );
+* // returns 2
+*/
+declare function countIfs<U = unknown>( x0: Collection<U> | AccessorArrayLike<U>, predicate0: Predicate<U>, x1: Collection<U> | AccessorArrayLike<U>, predicate1: Predicate<U> ): number;
```

Comment:
This isn't correct. `x0` may have type `U`, but `x1` should be allowed to have type `V`. In short, each array should be able to have its own type.